checker/jdk/index/src/java/nio/CharBuffer.java

```diff
@@ -0,0 +1,1514 @@
+ *
+
+*
+
+ *
+ * <p> This class implements the {@link CharSequence} interface so that
+ * character buffers may be used wherever character sequences are accepted, for
+ * example in the regular-expression package <tt>{@link java.util.regex}</tt>.
+ * </p>
+ *
+
+ *
+
+
+
+ *
+ * <p> Methods in this class that do not otherwise have a value to return are
+ * specified to return the buffer upon which they are invoked.  This allows
+ * method invocations to be chained.
+ *
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+ *
+ * The sequence of statements
+ *
+ * <blockquote><pre>
+ * cb.put("text/");
+ * cb.put(subtype);
+ * cb.put("; charset=");
+ * cb.put(enc);</pre></blockquote>
+ *
+ * can, for example, be replaced by the single statement
+ *
+ * <blockquote><pre>
+ * cb.put("text/").put(subtype).put("; charset=").put(enc);</pre></blockquote>
+ *
+
+ *
+ *
+ * @author Mark Reinhold
+ * @author JSR-51 Expert Group
+ * @since 1.4
+ */
+
+public abstract class CharBuffer
+    extends Buffer
+    implements Comparable<CharBuffer>, Appendable, CharSequence, Readable
+{
+
+    // These fields are declared here rather than in Heap-X-Buffer in order to
+    // reduce the number of virtual method invocations needed to access these
+    // values, which is especially costly when coding small buffers.
+    //
+    final char[] hb;                  // Non-null only for heap buffers
+    final @NonNegative int offset;
+    boolean isReadOnly;                 // Valid only for heap buffers
+
+    // Creates a new buffer with the given mark, position, limit, capacity,
+    // backing array, and array offset
+    //
+    CharBuffer(@GTENegativeOne @LessThan("#2 + 1") int mark, @NonNegative @LessThan("#3 + 1") int pos, @NonNegative @LessThan("#4 + 1") int lim, @NonNegative int cap,   // package-private
+                 char[] hb, @NonNegative int offset)
+    {
+        super(mark, pos, lim, cap);
+        this.hb = hb;
+        this.offset = offset;
+    }
+
+    // Creates a new buffer with the given mark, position, limit, and capacity
+    //
+    CharBuffer(@GTENegativeOne @LessThan("#2 + 1") int mark, @NonNegative @LessThan("#3 + 1") int pos,@NonNegative @LessThan("#4 + 1") int lim, @NonNegative int cap) { // package-private
+        this(mark, pos, lim, cap, null, 0);
+    }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+    /**
+     * Allocates a new char buffer.
+     *
+     * <p> The new buffer's position will be zero, its limit will be its
+     * capacity, its mark will be undefined, and each of its elements will be
+     * initialized to zero.  It will have a {@link #array backing array},
+     * and its {@link #arrayOffset array offset} will be zero.
+     *
+     * @param  capacity
+     *         The new buffer's capacity, in chars
+     *
+     * @return  The new char buffer
+     *
+     * @throws  IllegalArgumentException
+     *          If the <tt>capacity</tt> is a negative integer
+     */
+    public static CharBuffer allocate(int capacity) {
```

Comment:
`capacity` should be `@NonNegative`. When annotating a file for the annotated JDK, you should always look through the whole file and annotate every method, not just the ones you need.